### PR TITLE
Ensure we copy binaries and use glibc inside containers

### PR DESCRIPTION
**Pull request by tonyhb**
## Description

Even though we pass `-static` to `ld` during compilation, we're compiling against glibc instead of musl - so we switch out the base image for debian slim to fix containers using CGO.

We also need to change the files that we copy in using non-relative paths:

```
(git)-[fix/goreleaser-docker] % docker run --rm -ti 49445b0cbf30

    ____                            __
   /  _/___  ____  ____ ____  _____/ /_
   / // __ \/ __ \/ __ '/ _ \/ ___/ __/
 _/ // / / / / / / /_/ /  __(__  ) /_
/___/_/ /_/_/ /_/\__, /\___/____/\__/
                /____/

Build event-driven queues with zero infra.
Request features, get help, and chat with us: https://www.inngest.com/discord

Usage:
  inngest [command]

Available Commands:
  dev         Run the Inngest dev server
  help        Help about any command
  login       Logs in to your Inngest account
  serve       Start an Inngest service for self hosting
  version     Shows the inngest CLI version (saving time, it's: dev-)

Flags:
  -h, --help      help for inngest
      --json      Output logs as JSON.  Set to true if stdout is not a TTY.
      --prod      Use the production environment for the current command.
  -v, --verbose   Enable verbose logging.

Use "inngest [command] --help" for more information about a command.
```

## Type of change (choose one)
- [ ] Chore (refactors, upgrades, etc.)
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] Security fix (non-breaking change that fixes a potential vulnerability)
- [ ] Docs
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## Checklist
- [ ] I've linked any associated issues to this PR.
- [ ] I've tested my own changes.

*[Check our Pull Request Guidelines](https://github.com/inngest/inngest/blob/main/docs/PULL_REQUEST_GUIDELINES.md)*

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ dockers:
     - "inngest/inngest:{{ .Tag }}-amd64"
     build_flag_templates:
     - "--pull"
-    - "--build-arg=BIN=./build/inngest-linux-amd64"
```
Comment (djfarrelly) [inline .goreleaser.yml:24]: Checking the logs of the last release, `xgo` seems to run this command: which outputs into the `./build` dir:
```
go build -trimpath --mod=vendor '--ldflags= -s -w -X github.com/inngest/inngest/pkg/inngest/version.Version=v0.15.3 -X github.com/inngest/inngest/pkg/inngest/version.Hash=c75c0d51' -o /build/inngest-linux-amd64 ./cmd
```

https://github.com/inngest/inngest/blob/bee5ef1b1c1fd987cf7f981cddd902fa37642b02/.github/workflows/release.yml#L47

And the goreleaser seems to correctly copy it here: https://github.com/inngest/inngest/actions/runs/5557582337/jobs/10151552533#step:13:159 

Is there some context that I'm missing about the build or Github workflow? I'd love to more fully understand this change and how it works :)

**Review by djfarrelly — COMMENTED** [review]
Hey @tonyhb - thanks for this fix that replaces the need for #518. I have a question and if that all looks good then I can approve this for merge!

File: .goreleaser.yml (hunk 1 above)
Comment (tonyhb) [reply .goreleaser.yml:24]: Goreleaser copies from the build output, eg. the build folder, so we dont need to use relative paths.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -6,8 +6,47 @@ on:
       - "v*"
 
 jobs:
+  windows-build:
+    runs-on: windows-latest
+    staeps:
```
Comment (djfarrelly) [inline .github/workflows/release.yml:11]: @tonyhb I think this is a typo

**Pull request merged**